src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -293,6 +292,8 @@ void QMCCostFunctionBatched::checkConfigurations()
     RefVector<ParticleSet> p_list        = convertPtrToRefVector(p_ptr_list_);
```

Comment:
auto can make the code more brief.

Reply:
auto can also make the code harder to reason about types.  Maybe not so much in this case because of the function name.

Reply:
This would be better addressed in #2702, which touches those lines.

Reply:
Exactly. Thanks to the function name. No ambiguity here. It is like using auto with make_unique.

Reply:
Changed in #2702